### ISSUE TITLE
Ability to add more than one account of an exchange

### ISSUE THREAD

**Issue by lordi** [rotki/rotki#47]
Even thought this might be an issue that won't affect many people, I think it is important to include the possibility for it early on in the development process so that data models etc allow for it.

**Comment by rakanalh** [comment]
I also agree this should be done... at some point i created a different binance account to be able to "separate" funds... being able to import both accounts would be great.

**Comment by busera** [comment]
I also looking for multiple account support for one exchange. I have multiple sub-accounts at the Binance exchange.

**Comment by LefterisJP** [comment]
Tentatively putting in v1.9.0 milestone. May be easier than expected and may implement earlier. Or harder and take longer. Would need to see.

Naturally if any open source contributor would like to tackle this it would be really great.

**Comment by LefterisJP** [comment]
Just to get some clarification @lordi @rakanalh @busera @grooveuser your requirement is:

1.Be able to add multiple exchange accounts and have all balances, history, trades, deposit/withdrawals etc. bundled. So whenever you get your trades you will still see all your (let's say kraken) trades under the Kraken logo. But in reality they will be from many different kraken accounts.
2. Be able to add multiple exchange accounts and have all balances, history, trades, deposit/withdrawals managed separately. So whenever you get your trades let's say you will see all your (let's say kraken) trades under the different Kraken entries. Kraken account-1, kraken account-2 etc.

I understand the feature request as bullet point (1). Bullet point (2) is considerably much much harder for us to implement. What were you guys thinking?

**Comment by busera** [comment]
Option 1 would  be fantastic. 
Option 2 is something for the loooong run. 
However, having all accounts of an exchanged consolidated into one (Otption 1) will help a lot. i am interested in the overall performance of my accounts.

**Comment by kilrau** [comment]
+1 and option 1 suffices :+1:

**Comment by LefterisJP** [comment]
I am taking this up for v1.17.0 as discussed in the standup.

**Comment by fvanderelst** [comment]
Thank you for tackling this, I've found that problematic too.  I'll try to test.  Best wishes !

**Comment by LefterisJP** [comment]
Not fixed yet. A bit of frontend work by @kelsos is still needed. after #2780 got merged.

**Comment by LefterisJP** [comment]
Changes needed in the frontend:

- The string representation changes for 2 locations. for crypto.com -> cryptocom and for binance_us -> binance us
- Multiple api keys can now be added per exchange. And exchange credential entries can be edited. Not the actual credentials but the peripherals like exchange name, passphrase if existing, kraken account type if existing.
- kraken_account_type is removed from settings.
- To see the endpoint changes check the api docs in the PR: https://github.com/rotki/rotki/pull/2780/files
- TL;DR: Now an exchange is uniquely identified by location + name and you can also edit exchange credentials.
- e2e tests will also need fixing
